src/nats.c

```diff
@@ -879,6 +928,17 @@ nats_Open(int64_t lockSpinCount)
         s = natsNUID_init();
 
     if (s == NATS_OK)
+        s = natsMutex_Create(&(gLib.dlvWorkers.lock));
+    if (s == NATS_OK)
+    {
+        gLib.libHandlingMsgDeliveryByDefault = (getenv("NATS_DEFAULT_TO_LIB_MSG_DELIVERY") != NULL ? true : false);
+        gLib.dlvWorkers.maxSize = 2;
+        gLib.dlvWorkers.workers = NATS_CALLOC(gLib.dlvWorkers.maxSize, sizeof(natsMsgDlvWorker*));
+        if (gLib.dlvWorkers.workers == NULL)
+            s = NATS_NO_MEMORY;
+    }
```

Comment:
I wanted to use that as a backdoor for those like you that would want the behavior to always be "a global message delivery thread pool" as opposed to having each subscription creates its delivery thread, without having to bother creating and setting the option (through `natsOptions_UseGlobalMessageDelivery`). Incidentally, I am using this for the test suite and valgrind to verify that all tests can use that way of message delivery and still work properly without memory leak (which I have included in the Travis build matrix).
I will see how I can document this in doxygen.